python/tvm/topi/cuda/scatter.py

```diff
@@ -312,19 +313,18 @@ def gen_ir_4d(data, indices, updates, axis, out, update_func):
     out_ptr = ib.buffer_ptr(out)
     data_ptr = ib.buffer_ptr(data)
     with ib.new_scope():
-        i = te.thread_axis("blockIdx.x")
-        ib.scope_attr(i, "thread_extent", n)
-        j = te.thread_axis("blockIdx.y")
-        ib.scope_attr(j, "thread_extent", c)
-        k = te.thread_axis("blockIdx.z")
-        ib.scope_attr(k, "thread_extent", h)
+        fused = n * c * h * w
+        num_thread = int(tvm.target.Target.current(allow_none=False).max_num_threads)
+        num_blocks = ceil_div(fused, num_thread)
```

Comment:
:) I didn't think of this as an option when I did the scatter kernel, this was my first GPU kernel. Can we make this change on the other dimensions as well?

Reply:
I can, but I think a better approach is to have a reusable memcpy IR, to avoid writing the same boring kernels over and over again. I think memcpy and init-by-constant IR are generally useful.

Reply:
autotir can't come soon enough...

Reply:
Now all memcpy are done by the same IR.